src/main/java/com/guysfromusa/carsgame/control/GameEngine.java

```diff
@@ -29,30 +28,36 @@
 
     private final CarService carService;
 
+    private final CarMoveHandler carMoveHandler;
+
     @Inject
     public GameEngine(ActiveGamesContainer activeGamesContainer,
                       ApplicationEventPublisher applicationEventPublisher,
-                      CarService carService) {
+                      CarService carService,
+                      CarMoveHandler carMoveHandler) {
         this.activeGamesContainer = notNull(activeGamesContainer);
         this.applicationEventPublisher = notNull(applicationEventPublisher);
-        this.carService = carService;
+        this.carService = notNull(carService);
+        this.carMoveHandler = notNull(carMoveHandler);
     }
 
     @Async
-    public void handleMoves(List<Command> commands, String gameId) {
-        GameState gameState = activeGamesContainer.getGameState(gameId);
+    public void handleMoves(List<Command> commands, String gameName) {
+        GameState gameState = activeGamesContainer.getGameState(gameName);
 
-        //TODO for all commands calculate movements and collisions
+        commands.stream()
```

Comment:
the idea was that in one round we should move all the cars and check collisions. What is here is stream (which executes operations in sequence) and causing multicollision not possible because after collision car is removed so 3rd car will not see 1st and 2nd anymore.